sdk/servicebus/service-bus/src/util/utils.ts

```diff
@@ -180,3 +182,198 @@ export function getAssociatedReceiverName(
   }
   return receiverName;
 }
+
+/**
+ *  @ignore
+ * Helper utility to retrieve `string` value from given input,
+ * or undefined if not passed in.
+ * @param value
+ */
+export function getStringOrUndefined(value: any): string | undefined {
+  if (value == undefined) {
+    return undefined;
+  }
+  return value.toString();
+}
+
+/**
+ *  @ignore
+ * Helper utility to retrieve `integer` value from given string,
+ * or undefined if not passed in.
+ * @param value
+ */
+export function getIntegerOrUndefined(value: any): number | undefined {
+  if (value == undefined) {
+    return undefined;
+  }
+  const result = parseInt(value.toString());
+  return result == NaN ? undefined : result;
+}
+
+/**
+ *  @ignore
+ * Helper utility to retrieve `boolean` value from given string,
+ * or undefined if not passed in.
+ * @param value
+ */
+export function getBooleanOrUndefined(value: any): boolean | undefined {
+  if (value == undefined) {
+    return undefined;
+  }
+  return (
+    value
+      .toString()
+      .trim()
+      .toLowerCase() === "true"
+  );
+}
+
+/**
+ *  @ignore
+ * Helper utility to retrieve `JSON` value from given input,
+ * or undefined if not passed in. Treats empty string as invalid JSON
+ * @param value
+ */
+export function getJSONOrUndefined(value: any): any | undefined {
```

Comment:
I'm probably missing something, but this entire function feels like the following simplified version:

```js
// coerce null to undefined
if (value == undefined) return undefined;
// coerce empty strings and whitespace-only strings to undefined.
if (typeof value === 'string' && value.trim() === '') return undefined;

if (typeof value !== 'object') throw new Error(...);

return value;
```

Reply:
Which makes me wonder, why the special treatment for string (the second if-check)?

Reply:
Yes, this is because the service treats empty strings as valid JSON like value such as for authorization rules.

On suggested changes, yes - since JSON.parse() doesnt handle empty spaces well, just checking and excluding all types except JSON would work too.
To that point, we'll need to technically exclude `symbol` as well I think

```
There are 7 basic data types in JavaScript.

- `number` for numbers of any kind: integer or floating-point.
- `string` for strings. A string may have one or more characters, there’s no separate single-character type.
- `boolean` for true/false.
- `null` for unknown values – a standalone type that has a single value null.
- `undefined` for unassigned values – a standalone type that has a single value undefined.
- `object` for more complex data structures.
- `symbol` for unique identifiers.
```

JSON seems to be different from JS object.
For the XML serialization, the data needs to be in JSON format (I think this might further explain #5885)
Will update code and see if treating as JS object suffices, since the request gets converted to JSON anyway.

Reply:
Since JS object suffices, looks like this can be a simpler check of just `typeof value === "object"` - updated helper @ramya-rao-a

Reply:
Actually.. since `String` is an `object` type, the safer thing to do seems to be as what was done originally. Checking on the formatted string value and ensuring it starts with `{` or `[`
Thoughts? @ramya-rao-a @bterlson 
Pushing commit with these changes, would keep this unless there's still any concern with intent of helper.

Reply:
Updated helpers to be more clear and precise with the checks and purpose. Please take a look @bterlson @ramya-rao-a